src/icalendar/parser.py

```diff
@@ -428,37 +450,64 @@ def from_parts(
             return cls(f"{name};{params}:{values}")
         return cls(f"{name}:{values}")
 
-    def parts(self):
-        """Split the content line up into (name, parameters, values) parts."""
+    def parts(self) -> tuple[str, Parameters, str]:
+        """Split the content line up into (name, parameters, values) parts.
+
+        Properly handles escaping with backslashes and double-quote sections
+        to avoid corrupting URL-encoded characters in values.
+
+        Example with parameter::
+
+            DESCRIPTION;ALTREP="cid:part1.0001@example.org":The Fall'98 Wild
```

Comment:
Use explicit syntax highlighter.

```suggestion
        Example with parameter:
        
        .. code-block:: text

            DESCRIPTION;ALTREP="cid:part1.0001@example.org":The Fall'98 Wild
```